src/detect.h

```diff
@@ -1130,8 +1134,17 @@ typedef struct DetectEngineCtx_ {
     HashTable *non_pf_engine_names;
 
     const char *firewall_rule_file_exclusive;
+
+    /* user provided rate filter callbacks. */
+    SCDetectRateFilterFunc rate_filter_callback;
+
+    /* use provided data to be passed to rate_filter_callback. */
+    void *rate_filter_callback_arg;
```

Comment:
Need to be copied into new DetectEngineCtx on reload, or maybe move to master context, or global (for now).

Reply:
ah, nm my prev comment